### PR TITLE
feat: add pending tx queries

### PR DESCRIPTION
**Pull request by UncleSamtoshi**
- add pendingTransactions to Account and Wallet GQL objects
- add pendingTransactionsByAddress to Wallet objects

### CODE REVIEW

File: core/api/dev/apollo-federation/supergraph.graphql [hunk 1]
```diff
@@ -1960,6 +1972,25 @@ interface Wallet
...
+  """
+  pendingTransactions: [Transaction!]!
+
+  """
+  Pending OnChain transactions. When transactions
+  are confirmed they will receive a new id and be found in the transactions
+  list. Transactions are ordered anti-chronologically,
+  ie: the newest transaction will be first
+  """
+  pendingTransactionsByAddress(
```
Comment (dolcalmi) [inline core/api/dev/apollo-federation/supergraph.graphql:1990]: should not this be a filter of `pendingTransactions` ? cc @vindard @bodymindarts @nicolasburtey

File: core/api/src/app/wallets/get-pending-transactions-by-addresses.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { CouldNotFindError } from "@/domain/errors"
+
+import { WalletOnChainPendingReceiveRepository } from "@/services/mongoose"
+
+export const getPendingTransactionsForWalletsByAddresses = async ({
```
Comment (dolcalmi) [inline core/api/src/app/wallets/get-pending-transactions-by-addresses.ts:5]: I know wallets are passed from source but we must enforce in app layer that wallets belong the user/account (similar to logic in `getPendingOnChainTransactionsForAccountByWalletIds`)

File: core/api/src/graphql/admin/schema.graphql [hunk 3]
```diff
@@ -424,6 +429,11 @@ type UsdWallet implements Wallet {
 
   """An unconfirmed incoming onchain balance."""
   pendingIncomingBalance: SignedAmount!
+  pendingTransactions: [Transaction!]!
+  pendingTransactionsByAddress(
```
Comment (dolcalmi) [inline core/api/src/graphql/admin/schema.graphql:433]: how is this working if you need to pass wallet ids? did you test it?

File: core/api/src/graphql/public/types/object/consumer-account.ts [hunk 4]
```diff
@@ -206,6 +208,35 @@ const ConsumerAccount = GT.Object<Account, GraphQLPublicContextAuth>({
...
+      resolve: async (source, args) => {
+        let { walletIds } = args
+
+        if (!walletIds) {
+          const wallets = await WalletsRepository().listByAccountId(source.id)
+          if (wallets instanceof Error) {
+            throw mapError(wallets)
+          }
+          walletIds = wallets.map((wallet) => wallet.id)
+        }
```
Comment (dolcalmi) [inline core/api/src/graphql/public/types/object/consumer-account.ts:227]: this logic belongs to app layer (I know in some legacy code we use services from graphql layer but we should not introduce more debt on this)

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: core/api/src/graphql/admin/schema.graphql (hunk 3 above)
Comment (UncleSamtoshi) [reply core/api/src/graphql/admin/schema.graphql:433]: You dont need to pass wallet ids, this is already on the wallet object, so it uses the id of the wallet it is on.

File: core/api/src/app/wallets/get-pending-transactions-by-addresses.ts (hunk 2 above)
Comment (UncleSamtoshi) [reply core/api/src/app/wallets/get-pending-transactions-by-addresses.ts:5]: I dont necessarily agree with that. The function does what it says which is `getPendingTransactionsForWalletsByAddress`. It's different than the `getPendingOnChainTransactionsForAccountByWalletIds` which says its getting them for a specific account and therefore must verify that the account owns the wallets. I think this is a pretty common pattern across the app layer.

File: core/api/dev/apollo-federation/supergraph.graphql (hunk 1 above)
Comment (UncleSamtoshi) [reply core/api/dev/apollo-federation/supergraph.graphql:1990]: @bodymindarts reviewed this comment over standup and believes we should continue with this pattern

**Review by dolcalmi — APPROVED** [review]

**Pull request merged**